app/src/main/java/com/google/samples/apps/sunflower/compose/home/HomeScreen.kt

```diff
@@ -73,46 +65,56 @@ enum class SunflowerPage(
     PLANT_LIST(R.string.plant_list_title, R.drawable.ic_plant_list_active)
 }
 
+@OptIn(ExperimentalMaterial3Api::class, ExperimentalFoundationApi::class)
 @Composable
 fun HomeScreen(
     modifier: Modifier = Modifier,
     onPlantClick: (Plant) -> Unit = {},
-    onPageChange: (SunflowerPage) -> Unit = {},
-    onAttached: (Toolbar) -> Unit = {},
+    viewModel: PlantListViewModel = hiltViewModel()
 ) {
-    val activity = (LocalContext.current as AppCompatActivity)
+    val pagerState = rememberPagerState()
+    val scrollBehavior = TopAppBarDefaults.enterAlwaysScrollBehavior()
 
-    AndroidViewBinding(factory = HomeScreenBinding::inflate, modifier = modifier) {
-        onAttached(toolbar)
-        activity.setSupportActionBar(toolbar)
-        composeView.setContent {
-            HomePagerScreen(onPlantClick = onPlantClick, onPageChange = onPageChange)
+    Scaffold(
+        modifier = modifier.nestedScroll(scrollBehavior.nestedScrollConnection),
+        topBar = {
+            HomeTopAppBar(
+                pagerState = pagerState,
+                onFilterClick = { viewModel.updateData() },
+                scrollBehavior = scrollBehavior
+            )
         }
+    ) {
+        HomePagerScreen(
+            onPlantClick = onPlantClick,
+            pagerState = pagerState,
+            scrollBehavior = scrollBehavior,
+            modifier = Modifier.padding(it)
+        )
     }
 }
 
-@OptIn(ExperimentalFoundationApi::class)
+@OptIn(ExperimentalFoundationApi::class, ExperimentalMaterial3Api::class)
 @Composable
 fun HomePagerScreen(
     onPlantClick: (Plant) -> Unit,
-    onPageChange: (SunflowerPage) -> Unit,
+    pagerState: PagerState,
+    scrollBehavior: TopAppBarScrollBehavior,
     modifier: Modifier = Modifier,
     pages: Array<SunflowerPage> = SunflowerPage.values()
 ) {
-    val pagerState = rememberPagerState()
-
-    LaunchedEffect(pagerState.currentPage) {
-        onPageChange(pages[pagerState.currentPage])
-    }
 
     // Use Modifier.nestedScroll + rememberNestedScrollInteropConnection() here so that this
     // composable participates in the nested scroll hierarchy so that HomeScreen can be used in
     // use cases like a collapsing toolbar
-    Column(modifier.nestedScroll(rememberNestedScrollInteropConnection())) {
+    Column(modifier.nestedScroll(scrollBehavior.nestedScrollConnection)) {
```

Comment:
I think we can remove the `nestedScroll` modifier here entirely. That was mostly here for the interop (i.e. CoordinatorLayout containing a ComposeView). If you remove this, does the collapsing toolbar still work?